### ISSUE TITLE
`docker stack deploy` to Ubuntu hosted Docker CE [edge] fails with "Error response from daemon: page not found"

### ISSUE THREAD

**Issue by bloo** [docker/for-mac#1697]
### Expected behavior

The `docker stack deploy` command works from the Docker console, as launched by the Docker Edge for Mac OS utility.

### Actual behavior

The command is met with a `Error response from daemon: page not found` error and fails to deploy the stack.

### Information

```
Docker for Mac: version: 17.06.0-rc1-ce-mac13 (4cdec4294a50b2233146b09469b49937dabdebdd)
macOS: version 10.12.5 (build: 16F73)
logs: /tmp/CF28C655-B3E8-4D6E-B8A9-7755D8616C6A/20170606-143018.tar.gz
[OK]     db.git
[OK]     vmnetd
[OK]     dns
[OK]     driver.amd64-linux
[OK]     virtualization VT-X
[OK]     app
[OK]     moby
[OK]     system
[OK]     moby-syslog
[OK]     db
[OK]     env
[OK]     virtualization kern.hv_support
[OK]     slirp
[OK]     osxfs
[OK]     moby-console
[OK]     logs
[OK]     docker-cli
[OK]     menubar
[OK]     disk
```

Also, I ran + uploaded it with the **Diagnostic ID** *CF28C655-B3E8-4D6E-B8A9-7755D8616C6A*.

### Steps to reproduce the behavior

  1. Compose file: `web.yml`
```
version: '3.2'
services:
  web:
    image: nginx:alpine
```

  2. From Mac OS, run `docker stack deploy -c web.yml web`
  3. Watch it fail:
```
service web: Error response from daemon: page not found
```

If I run it from the server running the docker engine itself (so there's no version mismatch), it works. Here is the version mismatch - and attempted downgrade - from the Mac OS client:

```
$ docker version
Client:
 Version:      17.06.0-ce-rc1
 API version:  1.29 (downgraded from 1.30)
 Go version:   go1.8.1
 Git commit:   7f8486a
 Built:        Wed May 31 02:56:01 2017
 OS/Arch:      darwin/amd64

Server:
 Version:      17.05.0-ce
 API version:  1.29 (minimum version 1.12)
 Go version:   go1.7.5
 Git commit:   89658be
 Built:        Thu May  4 22:24:06 2017
 OS/Arch:      linux/amd64
 Experimental: true
```

**Comment by bloo** [comment]
Update - creating a bundle using `docker-compose bundle` and deploying the stack that way works - but is obviously not ideal.

**Comment by kinghuang** [comment]
It's related to the new config option in stacks. Covered in docker/cli#150, fixed in docker/cli#162, and on the list for cherry-picking into RC3.

**Comment by bloo** [comment]
@kinghuang thanks!
